frontend/src/routes/routes.tsx

```diff
@@ -0,0 +1,36 @@
+import { createBrowserRouter,redirect } from "react-router-dom";
+import { getTokens } from "../services/auth.service";
+import HomePage from "../pages/HomePage/HomePage";
+import LoginPage from "../pages/LoginPage/LoginPage";
+import NotFound from "../pages/NotFound/NotFound";
+import SignupPage from "../pages/SignupPage/SignupPage";
+import ErrorPage from "../pages/ErrorPage/ErrorPage";
+
+const authLoader = async () => {
+    const tokens = getTokens();
+    if (tokens.accessToken && tokens.refreshToken) {
+      return null;
+    }
+    return redirect("/login");
+};
+
+export const router = createBrowserRouter([
+    {
+      path: "/",
+      element: <HomePage />,
+      loader: authLoader,
+      errorElement: <ErrorPage/>,
+    },
+    {
+      path: "/login",
+      element: <LoginPage />,
+    },
+    {
+      path: "/signup",
+      element: <SignupPage />,
```

Comment:
**Problem** The routing configuration does not include a loader function for the /login and /signup routes. This means that these routes do not require authentication, allowing users to access them even if they are already authenticated.

**Fix**: Consider adding a loader function for the /login and /signup routes to handle cases where users are already authenticated. You can redirect authenticated users away from these pages to a different route, such as the homepage.

Reply:
got it